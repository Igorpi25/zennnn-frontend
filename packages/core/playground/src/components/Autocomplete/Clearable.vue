<template>
  <Autocomplete
    v-model="model"
    :items="items"
    :prepend-icon="ziSearch"
    clearable
    placeholder="Search..."
  >
  </Autocomplete>
</template>

<script setup>
import { ref } from 'vue'
import { ziSearch } from '@zennnn/icons'

const model = ref(null)
const items = Array.from(Array(10).keys()).map((key) => {
  const i = key + 1
  return {
    value: `item${i}`,
    text: `Item ${i}`,
    disabled: i === 3,
  }
})
</script>
