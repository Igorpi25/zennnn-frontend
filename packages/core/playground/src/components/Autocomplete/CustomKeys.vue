<template>
  <Autocomplete
    v-model="model"
    :items="items"
    item-text="name"
    item-value="id"
    placeholder="Select item"
  >
  </Autocomplete>
</template>

<script setup>
import { ref } from 'vue'

const model = ref(null)
const items = Array.from(Array(10).keys()).map(key => {
  const i = key + 1
  return {
    id: `item${i}`,
    name: `Item ${i}`,
  }
})
items.splice(3, 0, { divider: true })
</script>
