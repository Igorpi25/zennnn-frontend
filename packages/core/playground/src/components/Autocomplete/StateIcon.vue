<template>
  <Autocomplete
    v-model="model"
    :items="items"
    :rules="[rules.required]"
    placeholder="Select item"
    state-icon
  >
  </Autocomplete>
</template>

<script setup>
import { ref } from 'vue'

const model = ref(null)
const items = Array.from(Array(10).keys()).map(key => {
  const i = key + 1
  return {
    value: `item${i}`,
    text: `Item ${i}`,
  }
})
const rules = {
  required: v => !!v || 'Please select.',
}
</script>
