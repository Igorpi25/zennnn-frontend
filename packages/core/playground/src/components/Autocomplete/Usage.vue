<template>
  <Autocomplete
    v-model="model"
    :items="items"
    placeholder="Select item"
  >
  </Autocomplete>
</template>

<script setup>
import { ref } from 'vue'

const model = ref(null)
const items = Array.from(Array(10).keys()).map(key => {
  const i = key + 1
  return {
    value: `item${i}`,
    text: `Item ${i}`,
    disabled: i === 3,
  }
})
</script>
