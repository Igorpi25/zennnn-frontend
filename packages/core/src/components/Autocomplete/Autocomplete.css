@layer components {
  .autocomplete--is-menu-active::before {
    content: '';
    @apply absolute inset-x-0 bottom-0 pt-1;
    @apply bg-white dark:bg-gray-900;
  }

  .autocomplete-box::before {
    content: '';
    @apply absolute pointer-events-none inset-0 -top-10 rounded-t rounded-b-lg;
    @apply shadow-main-day dark:shadow-main-night;
  }
  .autocomplete-box--dense::before {
    @apply -top-8;
  }

  .listbox--searching .menu__item:not(.menu__item--disabled) {
    @apply text-gray-100;
  }
  .listbox--searching .menu__item--active:not(.menu__item--disabled) {
    @apply text-gray-200;
  }
  .listbox--searching
    .menu__item:not(.menu__item--disabled)
    .listbox__option__mask {
    @apply text-gray-900 dark:text-white;
  }
}
