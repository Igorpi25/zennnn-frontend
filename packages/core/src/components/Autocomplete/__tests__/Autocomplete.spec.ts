import { nextTick } from 'vue'
import { mount, VueWrapper } from '@vue/test-utils'
import { createI18n } from 'vue-i18n'
import { wait } from 'vue-supp'

import Autocomplete from '../Autocomplete'

jest.mock('../../../utils/uid')

describe('Autocomplete.ts', () => {
  let mountFunction: (options?: Record<string, unknown>) => VueWrapper<any>

  beforeEach(() => {
    document.body.innerHTML = '<div id="app"></div>'

    mountFunction = (options = {}) => {
      return mount(Autocomplete, {
        ...options,
        global: {
          plugins: [
            createI18n({
              silentTranslationWarn: true,
              silentFallbackWarn: true,
            }),
          ],
        },
      })
    }
  })

  // Render

  it('should render component and match snapshot', () => {
    const items: any[] = ['one', 'two', 'three']
    const wrapper = mountFunction({
      props: {
        label: 'Default',
        placeholder: 'Default',
        items,
      },
    })

    expect(document.querySelector('[data-popper-root]')?.outerHTML).toBe(
      undefined
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render items on focused and match snapshot', async () => {
    const items: any[] = ['one', 'two', 'three']
    const wrapper = mountFunction({
      props: {
        label: 'Default',
        placeholder: 'Default',
        items,
      },
    })

    await wrapper.get('input').trigger('focus')
    const itemsHtml = document.querySelector('[data-popper-root]')?.outerHTML

    expect(itemsHtml).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with disabled prop and match snapshot, should not open menu', async () => {
    const wrapper = mountFunction({
      props: {
        items: ['item1'],
        label: 'Default',
        placeholder: 'Default',
        disabled: true,
      },
    })

    await wrapper.get('input').trigger('focus')
    expect(document.querySelector('[data-popper-root]')?.outerHTML).toBe(
      undefined
    )

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with name, required, readonly props and match snapshot, should not open menu', async () => {
    const wrapper = mountFunction({
      props: {
        items: ['item1'],
        label: 'Default',
        placeholder: 'Default',
        name: 'text-field-name',
        required: true,
        readonly: true,
      },
    })

    await wrapper.get('input').trigger('focus')
    expect(document.querySelector('[data-popper-root]')?.outerHTML).toBe(
      undefined
    )

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with loading prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        label: 'Default',
        placeholder: 'Default',
        loading: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with singleLine prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        label: 'Default',
        placeholder: 'Default',
        singleLine: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with solo prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        items: ['item1'],
        label: 'Default',
        placeholder: 'Default',
        solo: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with dense prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        items: ['item1'],
        label: 'Default',
        placeholder: 'Default',
        dense: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with slots and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        items: ['item1'],
        prependIcon: 'prepend-icon',
        appendIcon: 'append-icon',
      },
      slots: {
        prepend: () => 'Prepend slot',
        append: () => 'Append slot',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  //

  it('should filter items', async () => {
    const items: any[] = ['one', 'two', 'three']
    const wrapper = mountFunction({
      props: {
        items,
        searchable: true,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.setValue('one')

    expect(document.querySelectorAll('[role="option"]').length).toBe(1)
  })

  // Interactions

  it('should select value on item click', async () => {
    const items: any[] = ['Item 1']
    const wrapper = mountFunction({
      props: {
        items,
        attach: false,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: ' ' })

    await wrapper.find('[role="option"]').trigger('click')

    expect(wrapper.emitted('update:modelValue')).toEqual([['Item 1']])
  })

  it('should open on Spacebar keydown', async () => {
    const items: any[] = ['Item 1']
    const wrapper = mountFunction({
      props: {
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: 'Esc' })

    expect(wrapper.vm.isMenuActive).toBe(false)

    await input.trigger('keydown', { key: ' ' })

    expect(wrapper.emitted().keydown.length).toBe(2)
    expect(wrapper.vm.isMenuActive).toBe(true)
  })

  it('should not open menu on focus and close on Esc keydown', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = ['Item 1', 'Item 2', 'Item 3']
    const wrapper = mountFunction({
      props: {
        modelValue: 'Item 2',
        items,
        openOnFocus: true,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')

    expect(wrapper.vm.isMenuActive).toBe(true)

    await input.trigger('keydown', { key: 'Esc' })

    expect(wrapper.emitted().keydown.length).toBe(1)
    expect(wrapper.vm.isMenuActive).toBe(false)
  })

  it('should activate first item on ArrowDown keydown', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = ['Item 1', 'Item 2', 'Item 3']
    const wrapper = mountFunction({
      props: {
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: ' ' })
    await input.trigger('keydown', { key: 'ArrowDown' })

    const item = document.querySelector('.menu__item--active') as HTMLElement

    expect(wrapper.emitted().keydown.length).toBe(2)
    expect(item.textContent).toBe('Item 1')
  })

  it('should activate last item on ArrowUp keydown', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = ['Item 1', 'Item 2', 'Item 3']
    const wrapper = mountFunction({
      props: {
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: ' ' })
    await input.trigger('keydown', { key: 'ArrowUp' })

    const item = document.querySelector('.menu__item--active') as HTMLElement

    expect(wrapper.emitted().keydown.length).toBe(2)
    expect(item.textContent).toBe('Item 3')
  })

  it('should activate selected item on ArrowDown/ArrowUp keydown', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = ['Item 1', 'Item 2', 'Item 3']
    const wrapper = mountFunction({
      props: {
        modelValue: 'Item 2',
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: 'ArrowDown' })
    await wait(20)
    expect(
      (document.querySelector('.menu__item--active') as HTMLElement)
        ?.textContent
    ).toBe('Item 2')

    await input.trigger('keydown', { key: 'Esc' })
    await nextTick()
    await input.trigger('keydown', { key: 'ArrowUp' })
    await wait(20)
    expect(
      (document.querySelector('.menu__item--active') as HTMLElement)
        ?.textContent
    ).toBe('Item 2')
  })

  it('should select value with keyboard', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = [{ text: 'Item 1', value: 1 }]
    const wrapper = mountFunction({
      props: {
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: ' ' })
    await input.trigger('keydown', { key: 'ArrowDown' })
    await input.trigger('keydown', { key: 'Enter' })

    expect(wrapper.emitted().keydown.length).toBe(3)
    expect(wrapper.emitted('update:modelValue')).toEqual([[1]])
  })

  it('should select active item and close menu on Tab keydown', async () => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn()

    const items: any[] = ['Item 1', 'Item 2', 'Item 3']
    const wrapper = mountFunction({
      props: {
        items,
      },
    })

    const input = wrapper.get('input')

    await input.trigger('focus')
    await input.trigger('keydown', { key: ' ' })
    await input.trigger('keydown', { key: 'ArrowDown' })
    await input.trigger('keydown', { key: 'Tab' })

    await nextTick()

    expect(wrapper.vm.isMenuActive).toBe(false)
    expect(wrapper.emitted().keydown.length).toBe(3)
    expect(wrapper.emitted('update:modelValue')).toEqual([['Item 1']])
  })
})
