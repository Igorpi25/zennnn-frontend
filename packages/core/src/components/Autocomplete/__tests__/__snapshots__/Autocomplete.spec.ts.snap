// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete.ts should render component and match snapshot 1`] = `
<div class="input autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with dense prop and match snapshot 1`] = `
<div class="input autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input input__input--dense autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with disabled prop and match snapshot, should not open menu 1`] = `
<div class="input input--disabled autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true" disabled="">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-gray-400 cursor-not-allowed" role="button" aria-hidden="false" tabindex="-1" disabled=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with loading prop and match snapshot 1`] = `
<div class="input autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with name, required, readonly props and match snapshot, should not open menu 1`] = `
<div class="input input--focused autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control autocomplete__control--is-active">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" name="text-field-name" required="" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true" readonly="">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with singleLine prop and match snapshot 1`] = `
<div class="input autocomplete">
  <!---->
  <!---->
  <div class="input__control input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with slots and match snapshot 1`] = `
<div class="input autocomplete">
  <!---->
  <!---->
  <div class="input__control input__control--has-prepend input__control--has-append autocomplete__control">
    <!---->
    <div class="flex items-center flex-shrink-0"><i class="icon text-gray-200 dark:text-gray-300" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
          <g fill="currentColor">
            <path d="prepend-icon"></path>
          </g>
        </svg></i>Prepend slot</div><input id="autocomplete-1" class="input__input autocomplete__input" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <div class="flex items-center flex-shrink-0">Append slot<i class="icon text-gray-200 dark:text-gray-300" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
          <g fill="currentColor">
            <path d="append-icon"></path>
          </g>
        </svg></i></div>
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render component with solo prop and match snapshot 1`] = `
<div class="input autocomplete">
  <!---->
  <!---->
  <div class="input__control input__control--solo input__control--has-append autocomplete__control">
    <!---->
    <!----><input id="autocomplete-1" class="input__input input__input--dense autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;

exports[`Autocomplete.ts should render items on focused and match snapshot 1`] = `undefined`;

exports[`Autocomplete.ts should render items on focused and match snapshot 2`] = `
<div class="input input--focused autocomplete"><label id="label-1" for="autocomplete-1" aria-hidden="false" class="label">
    <div class="label__wrapper">
      <div class="label__content__wrapper">
        <div class="label__content">Default</div>
      </div>
      <!---->
    </div>
  </label>
  <!---->
  <div class="input__control input__control--has-append autocomplete__control autocomplete__control--is-active">
    <!---->
    <!----><input id="autocomplete-1" class="input__input autocomplete__input" placeholder="Default" type="text" inputmode="text" autocomplete="off" aria-haspopup="true">
    <!---->
    <!---->
    <!----><button class="icon icon--link focus:outline-none origin-center transform transition-transform duration-200 ease-in-out text-blue-500 hover:text-blue-400 focus:text-blue-400" role="button" aria-hidden="false" tabindex="-1"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="m7.5 8.25 6 6-1.5 1.5-6-6 1.5-1.5zM18 9.75l-6 6-1.5-1.5 6-6 1.5 1.5z"></path>
        </g>
      </svg></button>
    <!---->
  </div><span class="menu"><!----><!----></span>
</div>
`;
